src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Datagrid.js

```diff
@@ -5,20 +5,36 @@ import Loader from '../../components/Loader';
 import Pagination from '../../components/Pagination';
 import DatagridStore from './stores/DatagridStore';
 import datagridStyles from './datagrid.scss';
-import TableAdapter from './adapters/TableAdapter';
+import datagridAdapterStore from './stores/DatagridAdapterStore';
 
 type Props = {
-    onRowEditClick?: (rowId: string | number) => void,
+    onItemClick?: (rowId: string | number) => void,
     store: DatagridStore,
+    views: Array<string>,
 };
 
 @observer
 export default class Datagrid extends React.PureComponent<Props> {
+    componentWillMount() {
+        this.props.views.forEach((viewName) => {
+            if (!datagridAdapterStore.has(viewName)) {
+                throw new Error(
```

Comment:
I think if we would do that (actually that's something I would have liked to discuss with @chirimoya, therefore the "Open for discussion" in my initial comment) I would say it goes into `componentWillReceiveProps` as well.

Reply:
ACK ... add to `componentWillReceiveProps`